File: third-party/node/node-zip/zip.cpp
// (c) Facebook, Inc. and its affiliates. Confidential and proprietary.

#include <nan.h>

#include "ZipAccess.h"
#include "ZipBuild.h"

namespace facebook {
namespace node_zip {

NAN_MODULE_INIT(init) {
  ZipAccess::init(target);
  ZipBuild::init(target);
}

} // namespace node_zip
} // namespace facebook

NODE_MODULE(zipstorage, facebook::node_zip::init)

File: third-party/node/node-zip/zip.js
/**
 * (c) Facebook, Inc. and its affiliates. Confidential and proprietary.
 *
 * @flow
 */

"use strict";

/*::
export type Entries = {
  files: Map<string, number>,
  links: Map<string, string>,
};

declare class ZipAccess {
  constructor(zipFile: string): ZipAccess;
  entries(): Entries;
  readFile(index: number, filenameDebug?: string): Buffer;
}

declare class ZipBuild {
  constructor(zipFile: string): ZipBuild;
  add(
    sourcePath: string,
    entryName: string,
    compressionLevel?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9,
  ): void;
  append(zipPath: string, prefix?: ?string): void;
  writeAndClose(): void;
}
*/

const os = require("os");
//$FlowFixMe: dynamic require needed to include platform-specific code
const native = require(`./${os.platform()}-${os.arch()}/zip.node`);

module.exports = (native /*: {
  ZipAccess: Class<ZipAccess>,
  ZipBuild: Class<ZipBuild>,
}*/);
